### PR TITLE
refactor(@ngtools/webpack): use cache for resource loader

### PR DESCRIPTION
**Pull request by filipesilva**

### CODE REVIEW

File: packages/@ngtools/webpack/src/resource_loader.ts [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
 const LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');
 const SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+const loaderUtils = require('loader-utils');
 
 
+interface CompilationResult {
+  hash: string;
+  outputName: string;
+  source: string;
+  newSource?: string;
```
Comment (hansl) [inline packages/@ngtools/webpack/src/resource_loader.ts]: Rename to `result` or `output` please.

File: packages/@ngtools/webpack/src/resource_loader.ts [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
+}
 
 export class WebpackResourceLoader {
+  private _parentCompilation: any;
   private _context: string;
   private _uniqueId = 0;
+  private _cache: { [k: string]: CompilationResult } = {};
```
Comment (hansl) [inline packages/@ngtools/webpack/src/resource_loader.ts]: Instead of a hash table, could you use a Map please?

Comment (filipesilva) [reply packages/@ngtools/webpack/src/resource_loader.ts]: Done

File: packages/@ngtools/webpack/src/resource_loader.ts (hunk 1 above)
Comment (filipesilva) [reply packages/@ngtools/webpack/src/resource_loader.ts]: Done

**Pull request merged**